react/features/base/connection/actions.any.ts

```diff
@@ -136,6 +137,38 @@ export function connectionFailed(
     };
 }
 
+/**
+ * Create an action for when the connection needs to be redirected to a visitor node or back to the main node.
+ *
+ * @param {JitsiConnection} connection - The {@code JitsiConnection} which
+ * failed.
+ * @param {string | undefined} vnode - The vnode to use or undefined if moving back to the main room.
+ * @param {string} focusJid - The focus jid to use.
+ * @param {string} username - The username to use.
+ * @public
+ * @returns {{
+ *     type: CONNECTION_FAILED,
+ *     connection: JitsiConnection,
+ *     vnode: string?,
+ *     focusJid: string,
+ *     username: string?
+ * }}
+ */
+export function connectionRedirected(
+        connection: Object,
+        vnode: String,
+        focusJid: String,
+        username: String) {
+
+    return {
+        type: CONNECTION_REDIRECTED,
```

Comment:
I'm wondering why don't we do this as async action and call all the logic from the middleware in it? I don't think this goes to the reducer at all.

Reply:
So basically you say to remove 
```
connection.addEventListener(
                JitsiConnectionEvents.CONNECTION_REDIRECTED,
                _onConnectionRedirected);
```
From here and move it to base/conference and to not dispatch action CONNECTION_REDIRECTED.

I don't like to move the connection logic from base/connection to base/conference.